src/OrchardCore.Modules/OrchardCore.Forms/Entities/Scripting/ValidateRuleMethod.cs

```diff
@@ -0,0 +1,242 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Scripting;
+
+namespace OrchardCore.Forms.Entities.Scripting
+{
+    public class ValidateRuleMethod : IGlobalMethodProvider

```

Comment:
This could directly execute the JS library or just execute the server-side validation directly in C# and not as `GlobalMethod`. If we use JS scripting then a validator could be to use free JS code too.